packages/job-manager/src/batch/pool-load-generator.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { injectable } from 'inversify';
+
+export interface PoolLoad {
+    activeTasks: number;
+    runningTasks: number;
+}
+
+export interface PoolMetricsInfo {
+    id: string;
+    maxTasksPerPool: number;
+    load: PoolLoad;
+}
+
+export interface PoolMetricsState {
+    lastPoolLoad?: PoolLoad;
+    lastTasksIncrementCount?: number;
+    processingSpeed?: number;
+}
+
+@injectable()
+export class PoolLoadGenerator {
+    private readonly poolState: PoolMetricsState = {};
+
+    public get processingSpeed(): number {
+        return this.poolState.processingSpeed;
+    }
+
+    public getTasksIncrementCount(poolMetricsInfo: PoolMetricsInfo, activeToRunningTasksRatio: number): number {
+        // No last pool state available. Use default task increment count.
+        if (this.poolState.lastTasksIncrementCount === undefined) {
+            this.poolState.lastPoolLoad = poolMetricsInfo.load;
+            this.poolState.processingSpeed = 0;
+            // account for existing active tasks in a pool
+            this.poolState.lastTasksIncrementCount =
+                poolMetricsInfo.maxTasksPerPool * activeToRunningTasksRatio - poolMetricsInfo.load.activeTasks;
+
+            return this.poolState.lastTasksIncrementCount;
+        }
+
+        // Calculate processing speed since last tasks increment.
+        this.poolState.processingSpeed =
+            this.poolState.lastTasksIncrementCount - (poolMetricsInfo.load.activeTasks - this.poolState.lastPoolLoad.activeTasks);
```

Comment:
>lastTasksIncrementCount [](start = 27, length = 23)

may be rename this to activeTasksFromLastIncrement? 
& set this value to zero if activeTasksFromLastIncrement is -ve?